### ISSUE TITLE
Error message could be clearer

### ISSUE THREAD

**Issue by tacman** [meilisearch/meilisearch#4392]
**Describe the bug**

     "Attribute `id` is not sortable. This index does not have configured sortable attributes.",

**Expected behavior**

I would LOVE to see the name of the index in this message.

     "Attribute `id` is not sortable. index 'Author' does not have configured sortable attributes.",

In particular, I _do_ have sortable attributes:

http://127.0.0.1:7700/indexes/Author/settings

![image](https://github.com/meilisearch/meilisearch/assets/619585/8cb2c9ef-0f13-42c9-8008-a58f938f4e65)

Perhaps the real issue is that the id is the primary key?  More likely I'm using the wrong index, but I can't see what that is.  A link to the settings would be even more awesome!

**Comment by tacman** [comment]
Perhaps the change could be make in the meilisearch php library?

**Comment by curquiza** [comment]
We would rather handle this at the source (engine) than changing things in 10 library

**Comment by tacman** [comment]
Makes sense.  In addition to the index, I guess it'd be helpful to have all the details.  Sometimes I run meilisearch in docker, and in different projects it listens on different ports, and sometimes calls a remote server.  So in addition to the index, I'd want to know the server and port.

**Comment by curquiza** [comment]
We would rather not prioritize the port addition to solve this issue -> let's focus on the index name as a first step. From a product perspective, it does not seem relevant to us to have this information in the error, so we need more product discussion before validating it. We are a small team so we are careful about everything we add because can be tedious to update and maintain

Btw, as a workaround, in the meantime you can have all the information in the URL you try to reach: index name and port

**Comment by cisco877** [comment]
I would like to see if I can work on this issue, would that be ok?

**Comment by curquiza** [comment]
Hello @cisco877 

Thanks for your interest in this project 🔥 You are definitely more than welcome to open a PR for this!

For your information, we prefer not assigning people to our issues because sometimes people ask to be assigned and never come back, which discourages the volunteer contributors from opening a PR to fix this issue.
We will accept and merge the first PR that fixes correctly and well implements the issue following our [contributing guidelines](../blob/main/CONTRIBUTING.md).   

We are looking forward to reviewing your PR 😊

**Comment by cisco877** [comment]
Hi @curquiza thanks for the reply, I understand, makes sense, I've been working on it yesterday, following the idea of propagating the index name from the route, I've tried to add index_uid to SearchQuery, so that I can have the index_uid in meilisearch/milli/src/search/new/mod.rs:709, but now I have SearchQuery and SearchQueryWithConfig and now both have the index_uid? Does that make sense now?

**Comment by postmeback** [comment]
Can someone write me the steps to replicate the issue?
Maybe I can give it a try.

**Comment by irevoire** [comment]
Hey, you should create an index with the method of your choice: https://www.meilisearch.com/docs/reference/api/indexes#create-an-index
For example in terminal you would run:
```
curl \
  -X POST 'http://localhost:7700/indexes' \
  -H 'Content-Type: application/json' \
  --data-binary '{
    "uid": "movies",
    "primaryKey": "id"
  }'
```

And then do a search with a filter on any field https://www.meilisearch.com/docs/reference/api/search#search-in-an-index-with-post
In terminal again it could be this request:
```
curl \
  -X POST 'http://localhost:7700/indexes/movies/search' \
  -H 'Content-Type: application/json' \
  --data-binary '{ "filter": "title = Kefirausaurus" }'
```

And this is the response you'll receive:
```json
{
  "message": "Attribute `title` is not filterable. This index does not have configured filterable attributes.\n1:6 title = Kefirausaurus",
  "code":"invalid_search_filter",
  "type":"invalid_request",
  "link":"https://docs.meilisearch.com/errors#invalid_search_filter"
}
```

**Comment by tacman** [comment]
@irevoire Yes, but I have hundreds of indexes.  "This index" is ambiguous, it'd be trivial to replace that generic term with the index name, so I knew exactly which index was causing the problem.  Especially since you can prefix the indexes when in development, so it's a lot of work to hunt down just the name of the index.

**Comment by irevoire** [comment]
@tacman Yes, I was just giving a reproducer for @postmeback, who wants to work on it

**Comment by tacman** [comment]
Got it, thanks!

**Comment by airycanon** [comment]
@irevoire Hi, as you said in https://github.com/meilisearch/meilisearch/pull/4414#pullrequestreview-1888056484, we would prefer to added the index name to every milli error message by implementing From<milli::Error> trait.

If I use the manual implementation of From<milli::Error>, do I need to add the index_name to the Error, but then I would not be able to pass the index name to the internal error messages of milli::Error?

**Comment by irevoire** [comment]
Hey @airycanon, thanks for looking into it!

> but then I would not be able to pass the index name to the internal error messages of milli::Erro

I’m not sure I understood this sentence.

I think what you should do is to remove the `From` `milli::Error` implementation of the `index_scheduler::Error` and instead do it by hand.
Also, you should update the error to something like that:
```rust
#[error("Index {index}: {error}")]
Milli { index: String, error: milli::Error }
```

I cannot guarantee you that it’ll be possible everywhere, but _I think_ it should work.

**Comment by airycanon** [comment]
@irevoire Got it. I will give it a try, and thanks for your patience.

**Comment by irevoire** [comment]
Hey @airycanon,

Yes, I think you’re right and you should do that!

**Comment by airycanon** [comment]
@irevoire  Since MeilisearchHttpError is converted to ResponseError, I found that I need to modify many functions that return ResponseError to add an index_uid parameter and pass it to MeilisearchHttpError. What do you think about this alternative approach?
```rust
trait ErrorWithIndex<T> {
    fn with_index(self, index_uid: &String) -> Result<T, ResponseError>;
}

impl<T> ErrorWithIndex<T> for Result<T, ResponseError> {
    fn with_index(self, index_uid: &String) -> Result<T, ResponseError> {
        self.map_err(|mut error| {
            error.message = format!("Index {}: {}",index_uid, error.message);
            error
        })
    }
}
```

**Comment by irevoire** [comment]
Hey @airycanon.

> What do you think about this alternative approach?

I can be wrong because I didn’t look at the code like you did, but I feel like this is more error-prone in the sense that we might forget to push the index name in a lot of place since it’s not related to the milli error anymore 🤔 

> I found that I need to modify many functions that return ResponseError to add an index_uid parameter and pass it to MeilisearchHttpError

And about that, aren’t you supposed to modify the exact same number of functions in both case? But in one case you get a compiler error and in the other case you need to find which function you must modify? 
If it’s the case, then I would prefer if we stick to the first approach and make sure we **cannot** forget to specify the index name.
If not, please let me know, I may have overlooked something 🥶

**Comment by curquiza** [comment]
Closed by #5056, will be in v1.12
